Frontend/Gmfctn/src/app/modules/+auth/components/sign-in/sign-in.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { finalize, take } from 'rxjs/operators';
+import { AuthenticateService } from '../../services/authenticate.service';
+
+@Component({
+  selector: 'app-sign-in',
+  templateUrl: './sign-in.component.html',
+  styleUrls: ['./sign-in.component.scss'],
+})
+export class SignInComponent implements OnInit, OnDestroy {
+  @Output() isSigned = new EventEmitter<boolean>();
+  state = '';
+  loading = false;
+  subscribtion$: Subscription = new Subscription;
+  subscribtionForCorrectness$: Subscription = new Subscription;
+  userForm: FormGroup = new FormGroup({});
+  user: {userName: string, password: string} = {userName: '', password: ''};
+
+
+  constructor(private readonly fb: FormBuilder, private authenticateService: AuthenticateService, private router: Router) { }
+
+  ngOnInit(): void {
+    this.subscribtionForCorrectness$ = this.authenticateService
+                                      .isErrorInAuthorization$
+                                      .subscribe( val => this.state = val ? 'Username or password are incorrect' : '' );
```

Comment:
Not sure what this statement should be doing but it's probably not)
`this.state = val ...` will be evaluated to the boolean equivalent of the val, which means the options will be conditionally called, but these strings aren't assigned to something, so it'll be evaluated and skipped

Reply:
If the interseptor caught some error while authorization, it would set isErrorInAuthorization in auth service as "true", and in this way after guard will redirect user to this route again, the component will show errormessage, else this variable will be set as empty string, and *ngIf wouldn`t invoke.